custom/src/main/java/com/splunk/opentelemetry/servicename/WebSphereServiceNameDetector.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.splunk.opentelemetry.servicename;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+class WebSphereServiceNameDetector extends AppServerServiceNameDetector {
+  private static final Logger log = LoggerFactory.getLogger(WebSphereServiceNameDetector.class);
+
+  WebSphereServiceNameDetector(ResourceLocator locator) {
+    super(locator, "com.ibm.wsspi.bootstrap.WSPreLauncher", true);
+  }
+
+  @Override
+  boolean isValidAppName(String name) {
+    // query.ear is bundled with websphere
+    return !"query.ear".equals(name);
+  }
+
+  @Override
+  Path getDeploymentDir() {
+    // not used
+    return null;
+  }
+
+  @Override
+  String detect() throws Exception {
+    if (serverClass == null) {
+      return null;
+    }
+
+    String programArguments = System.getProperty("sun.java.command");
+    log.debug("Started with arguments '{}'.");
+    if (programArguments == null) {
+      return null;
+    }
+
+    Pattern pattern =
+        Pattern.compile(
+            "com\\.ibm\\.wsspi\\.bootstrap\\.WSPreLauncher (.*) com\\.ibm\\.ws\\.runtime\\.WsServer (.+) ([^ ]+) ([^ ]+) ([^ ]+)");
+    Matcher matcher = pattern.matcher(programArguments);
+    if (!matcher.matches()) {
+      log.debug("Failed to parse arguments.");
+      return null;
+    }
+
+    // in docker image it is /opt/IBM/WebSphere/AppServer/profiles/AppSrv01/config
+    Path configDirectory = Paths.get(matcher.group(2));
+    if (!Files.exists(configDirectory)) {
+      log.debug("Missing configuration directory '{}'.");
+      return null;
+    }
+
+    String cell = matcher.group(3);
+    String node = matcher.group(4);
+    String server = matcher.group(5);
+
+    log.debug(
+        "Parsed arguments: cell '{}', node '{}', server '{}', configuration directory '{}'.",
+        cell,
+        node,
+        server,
+        configDirectory);
+
+    // construct installedApps directory path based on the config path
+    // in docker image it is
+    // /opt/IBM/WebSphere/AppServer/profiles/AppSrv01/installedApps/DefaultCell01
+    Path cellApplications = configDirectory.resolve("../installedApps/" + cell);
```

Comment:
I believe the installedApps location is configurable so there could be installations where this is not the correct path.

Reply:
Can you add a comment that mentions it? I think it's fine to hard-code it for now, but it would be nice to have a pointer in case there're any issues/requests about that.